compiler/src/dotty/tools/dotc/printing/Formatting.scala

```diff
@@ -168,7 +168,12 @@ object Formatting {
       case param: TypeParamRef =>
         s"is a type variable${addendum("constraint", ctx.typeComparer.bounds(param))}"
       case sym: Symbol =>
-        s"is a ${ctx.printer.kindString(sym)}${sym.showExtendedLocation}${addendum("bounds", sym.info)}"
+        val info =
+          if (ctx.gadt.bounds.contains(sym))
+            sym.info & ctx.gadt.bounds(sym)
+          else
+            sym.info
```

Comment:
micro optimisation:
```scala
val bounds = ctx.gadt.bounds(sym)
val info = if (bounds != null) sym.info & bounds else sym.info
```

Reply:
I don't think micro-optimizing error messages is worth it.